File: 2017/lua/day13.lua
scanners = {}
for line in io.lines("13.txt") do
    local depth, range = string.match(line, "(%d+): (%d+)");
    table.insert(scanners, {depth = tonumber(depth), range = tonumber(range)})
end

severity = 0
for _, v in ipairs(scanners) do
    if v.depth % (2 * (v.range - 1)) == 0 then severity = severity + v.depth * v.range end
end

delay = 0
repeat
    delay = delay + 1
    gotcha = false
    for _, v in ipairs(scanners) do
        if (v.depth + delay) % (2 * (v.range - 1)) == 0 then gotcha = true break end
    end
until not gotcha

print(string.format("Part 1: %d", severity))
print(string.format("Part 2: %d", delay))

File: 2017/kotlin/day13.kt
import java.io.File

class SecurityScanner(val depth: Int, val range: Int)

fun main() {
    val firewall = File("../__in/13.txt").readLines()
        .map { line -> line.split(": ").map { it.toInt() } }
        .map { SecurityScanner(it[0], it[1]) }

    var severity = 0
    for (layer in firewall) severity += if (layer.depth % (2 * (layer.range - 1)) == 0) layer.depth * layer.range else 0
    
    var delay = 0
    do {
        var caught = false
        delay++
        for (layer in firewall) {
            caught = (delay + layer.depth) % (2 * (layer.range - 1)) == 0
            if (caught) break
        }
    } while (caught)

    println("Part 1: $severity")
    println("Part 2: $delay")
}